### PR TITLE
suppport for passing multiple cluster nodes

### PR DESCRIPTION
**Pull request by anshulrvce**
### Description
connect to multiple cassandra cluster nodes by passing contactPoints as array in the config.

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch?

File: README.md [hunk 1]
```diff
@@ -13,24 +13,24 @@ In your application root directory, enter this command to install the connector:
 npm install loopback-connector-cassandra --save
 ```
 
-This installs the module from npm and adds it as a dependency to the application's `package.json` file.
+This installs the module from npm and adds it as a dependency to the application's `package.json` file.
```
Comment (kjdelisle) [inline README.md]: Noticing a bunch of seemingly unnecessary diffs here...are these tabs and not spaces?

File: README.md [hunk 2]
```diff
@@ -13,24 +13,24 @@ In your application root directory, enter this command to install the connector:
...
 
 If you create a Cassandra data source using the data source generator as described below, you don't have to do this, since the generator will run `npm install` for you.
 
 ## Creating a Cassandra data source
 
-Use the [Data source generator](http://loopback.io/doc/en/lb3/Data-source-generator.html) to add a Cassandra data source to your application.  Select `Cassandra` connector as follows:
+Use the [Data source generator](http://loopback.io/doc/en/lb3/Data-source-generator.html) to add a Cassandra data source to your application.  Select `Cassandra` connector as follows:
```
Comment (kjdelisle) [inline README.md]: Same thing, unnecessary whitespace/char changes.

File: README.md [hunk 3]
```diff
@@ -13,24 +13,24 @@ In your application root directory, enter this command to install the connector:
...
-  MongoDB (supported by StrongLoop) 
-  MySQL (supported by StrongLoop) 
+? Select the connector for mycass:
+  IBM Cloudant DB (supported by StrongLoop)
+  IBM DB2 for z/OS (supported by StrongLoop)
+  IBM WebSphere eXtreme Scale key-value connector (supported by StrongLoop)
+❯ Cassandra (supported by StrongLoop)
+  Redis key-value connector (supported by StrongLoop)
+  MongoDB (supported by StrongLoop)
+  MySQL (supported by StrongLoop)
```
Comment (kjdelisle) [inline README.md]: Undo these whitespace changes in the above diff section.

File: README.md [hunk 4]
```diff
@@ -42,8 +42,8 @@ $ lb datasource
 Connector-specific configuration:
 ? host: localhost
 ? port: 9042
-? user: 
-? password: 
+? user:
+? password:
```
Comment (kjdelisle) [inline README.md]: Undo these whitespace changes

File: README.md [hunk 5]
```diff
@@ -66,12 +66,27 @@ The entry in the application's `/server/datasources.json` will look like this:
   "connector": "cassandra"
 }
 ```
+For passing array of cluster nodes as contact points edit `datasources.json` as below:
```
Comment (kjdelisle) [inline README.md]: I'd rephrase this to be a bit clearer, something like:
```
To pass an array of cluster nodes as contact points, add a
`contactPoints` key to your `datasources.json` along with
an array of cluster names.
```

File: README.md [hunk 6]
```diff
@@ -66,12 +66,27 @@ The entry in the application's `/server/datasources.json` will look like this:
...
+  "connector": "cassandra"
+}
+```
 
 Edit `datasources.json` to add any other additional properties supported by [`cassandra-driver`](https://github.com/datastax/nodejs-driver).
 
 ## Type mappings
 
-See [LoopBack types](http://loopback.io/doc/en/lb3/LoopBack-types.html) for details on LoopBack's data types.
+See [LoopBack types](http://loopback.io/doc/en/lb3/LoopBack-types.html) for details on LoopBack's data types.
```
Comment (kjdelisle) [inline README.md]: remove whitespace changes

File: README.md [hunk 7]
```diff
@@ -84,7 +99,7 @@ In addition to the standard data types such as String, Boolean, and Number, seve
       <th>Cassandra Type</th>
     </tr>
   </thead>
-  <tbody>    
+  <tbody>
```
Comment (kjdelisle) [inline README.md]: remove whitespace changes

File: README.md [hunk 8]
```diff
@@ -206,7 +221,7 @@ Note that `clusteringKeys` is defined as an array because the order of the sorti
       <th>userId</th>
     </tr>
   </thead>
-  <tbody>    
+  <tbody>
```
Comment (kjdelisle) [inline README.md]: remove whitespace changes

File: lib/cassandra.js [hunk 9]
```diff
@@ -299,7 +301,7 @@ Cassandra.prototype.count = function(model, where, options, cb) {
 
 Cassandra.prototype.generateMissingDefaults = function(model, data) {
   var props = this.getModelDefinition(model).properties;
- 
+
```
Comment (kjdelisle) [inline lib/cassandra.js]: Remove unrelated formatting change here.

File: test/generateOptions.test.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
+const rewire = require('rewire');
```
Comment (kjdelisle) [inline test/generateOptions.test.js]: We should not use `rewire`. The author(s) of the package have said that their module is not compatible with faking modules that make use of `const` (https://github.com/jhnns/rewire):

>Please note: The current version of rewire is not compatible with const or babel. See Limitations.

Comment (kjdelisle) [reply test/generateOptions.test.js]: If you'd like to use a mock library, my personal recommendation would be [Sinon](http://sinonjs.org/).

File: test/generateOptions.test.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+            keyspace: 'test'
+        };
+        const expectedContactPointCount = 1;
+        const abc = generateOptions(settings);
+        const actualContactPointCount = abc.contactPoints.length;
+        (actualContactPointCount).should.be.equal(expectedContactPointCount);
+        done();
+    });
+
+});
```
Comment (kjdelisle) [inline test/generateOptions.test.js]: missing trailing newline here

File: test/generateOptions.test.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
+const rewire = require('rewire');
+const cassandraLib = rewire('../lib/cassandra.js');
+const generateOptions = cassandraLib.__get__('generateOptions');
+var should = require('should');
```
Comment (kjdelisle) [inline test/generateOptions.test.js]: use `const` here as well

**Comment by kjdelisle** [comment]
@anshulrvce Oh, almost forgot, there's a typo in the commit message (support has three "p"s).
If you `git commit --amend` and `git push <whatever you named our remote> +patch-4` that should work nicely.
